FlowCrypt/Controllers/Settings/Backup/Backups Option Scene/BackupOptionsViewController.swift

```diff
@@ -35,17 +35,24 @@ final class BackupOptionsViewController: ASDKViewController<TableNode> {
     private let backupService: BackupServiceType
     private let service: ServiceActor
     private let userId: UserId
+    private let appContext: AppContext
 
     init(
+        appContext: AppContext,
         decorator: BackupOptionsViewDecoratorType = BackupOptionsViewDecorator(),
-        backupService: BackupServiceType = BackupService(),
+        backupService: BackupServiceType? = nil,
         backups: [KeyDetails],
         userId: UserId
     ) {
+        self.appContext = appContext
         self.decorator = decorator
         self.backups = backups
-        self.backupService = backupService
+        let backupService = backupService ?? BackupService(
+            backupProvider: appContext.getRequiredMailProvider().backupProvider,
+            messageSender: appContext.getRequiredMailProvider().messageSender
+        )
```

Comment:
That would be great. But I have found that this may mean we would need to include `AppContext` in test target which is not desirable (because then everything gets included from there).

So I'm thinking I may instead add `getBackupProvider()` method to AppContext. It's less clean though.

I'll fiddle with it some more, see if I can add AppContext without adding everything (router / delegate / and with that all VCs.. not desired in tests)

Reply:
See here https://github.com/FlowCrypt/flowcrypt-ios/pull/1117/commits/e786059bb5492fe8f6d6d686df5871cee185fdc3

I know it's not what you wanted, but I'm unsure how to achieve what you proposed while keeping test target happy. We could file it as another issue, as what you proposed is nicer.

Reply:
Updated code looks better 👍
Another possible solution is to pass only `MailProvider` (as it's already included in test target), not full `AppContext`:
```swift
init(mailProvider: MailProvider) {
    self.backupProvider = mailProvider.backupProvider
    self.messageSender = mailProvider.messageSender
}
```
But I think your current solution is good enough.